### PR TITLE
create individual swagger.json for admin and public api

### PR DESCRIPTION
**Pull request by shabd67**
# Resolves

[SALUS-1019](https://jira.rax.io/browse/SALUS-1019)

# What

Creating swagger.json for public and admin apis to be uploaded to salus_www bucket along with index.html

### CODE REVIEW

**Comment by shabd67** [comment]
considering to add below code in cloud build to upload created swagger.json file to "salus_www" bucket.

`  - name: 'gcr.io/google.com/cloudsdktool/cloud-sdk'
    args: ['gsutil', 'cp', 'target/generated/swagger/public/swagger.json', 'gs://salus_www/monitor_management/public/']
  - name: 'gcr.io/google.com/cloudsdktool/cloud-sdk'
    args: ['gsutil', 'cp', 'target/generated/swagger/admin/swagger.json', 'gs://salus_www/monitor_management/admin/'`

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java [hunk 1]
```diff
@@ -93,15 +92,23 @@ public static void main(String[] args) throws Exception {
...
         });
 
         // generate the admin swagger json
         adminRoot.set("paths", adminPathNode);
-        mapper.writeValue(new java.io.File(args[0]+"/adminConvertedOutput.json"), (JsonNode)adminRoot);
+        mapper.writeValue(new java.io.File(args[0],"adminConvertedOutput.json"), (JsonNode)adminRoot);
+
+        File adminDir = new File(args[0],"admin");
+        adminDir.mkdirs();
+        mapper.writeValue(new java.io.File(adminDir, "swagger.json"), (JsonNode)adminRoot);
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:104]: I like the new file/dir structure you introduced; however, I'm wondering if `adminConvertedOutput.json` and `convertedOutput.json` (below) need to be kept. They'll have the same content as the respective `swagger.json`, right?

Comment (shabd67) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:104]: convertedOutput.json file is being used to create index.html file by swagger-codegen-maven-plugin plugin. And its content is like `/api/rebalance-zone/**` while admin/swagger.json file content would be like `/api/admin/rebalance-zone/**`

Comment (itzg) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:104]: You could then change the plugin execution to read from `admin/swagger.json` at

https://github.com/racker/salus-telemetry-monitor-management/blob/master/pom.xml#L337

...or not introduce any changes at all and have the cloud build step be
```
    args: ['gsutil', 'cp', 'target/generated/swagger/adminConvertedOutput.json', 'gs://salus_www/monitor_management/admin/'
```

Comment (itzg) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:104]: I'm not saying to combine the public and admin logic. I'm saying lines 100 and 104 seem to be creating two files with the same content. Why are two, duplicate files needed? Both lines are doing:
```
mapper.writeValue(...., (JsonNode)adminRoot);
```

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java [hunk 2]
```diff
@@ -93,15 +92,23 @@ public static void main(String[] args) throws Exception {
...
 
         // generate the public swagger json
         publicRoot.set("paths", publicPathNode);
-        mapper.writeValue(new java.io.File(args[0]+"/convertedOutput.json"), (JsonNode)publicRoot);
+        mapper.writeValue(new java.io.File(args[0], "convertedOutput.json"), (JsonNode)publicRoot);
+
+        File publicDir = new File(args[0],"public");
+        publicDir.mkdirs();
+        mapper.writeValue(new java.io.File(publicDir, "swagger.json"), (JsonNode)publicRoot);
```
Comment (itzg) [inline src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:112]: ...and separately, lines 108 and 112 seem to be creating two files with the same content. Both lines are doing:
```
mapper.writeValue( ....., (JsonNode)publicRoot);
```

Comment (shabd67) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:112]: yes, I got that @itzg . I will change that accordingly. but my question here https://github.com/racker/salus-tools/pull/42#discussion_r498332781 was a different one. its for distribution of public and Admin APIs.

**Comment by shabd67** [comment]
> Just to be clear @shabd67, why is the java code being changed at all. What is being accomplished with those changes?

that java code is responsible for creating different json input files for creating index.html files. I am just tweaking that logic to create different swagger.json as well.

File: src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java (hunk 2 above)
Comment (itzg) [reply src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java:112]: Separate json files are already being created here

https://github.com/racker/salus-tools/blob/eaa42f981b053e4d02c1f45cd452a92d18dbd55b/src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java#L101

and

https://github.com/racker/salus-tools/blob/eaa42f981b053e4d02c1f45cd452a92d18dbd55b/src/main/java/com/rackspace/salus/salus_tools/converter/SwaggerJsonConverter.java#L105

**Review by itzg — APPROVED** [review]

**Review by GeorgeJahad — APPROVED** [review]

**Pull request merged**